example/remove/t8_example_empty_trees.cxx

```diff
@@ -0,0 +1,189 @@
+
+  t8_forest_ref (forest);
+  t8_forest_t         forest_adapt;
+  t8_forest_init (&forest_adapt);
+  t8_forest_set_adapt (forest_adapt, forest, t8_adapt_remove, 0);
+  t8_forest_set_partition (forest_adapt, NULL, 0);
+  t8_forest_set_user_data (forest_adapt, &empty_tree);
+  t8_forest_commit (forest_adapt);
+
+  std::string vtuname_adapt (*vtuname);
+  vtuname_adapt = vtuname_adapt + "_adapted";
+  t8_forest_write_vtk (forest_adapt, vtuname_adapt.c_str ());
+  t8_debugf ("Output to %s\n", vtuname_adapt.c_str ());
+
+  t8_productionf ("The initial uniform forest:\n"
+                  "\tfirst_local_tree: %li\n"
+                  "\tlast_local_tree:  %li\n"
+                  "\tlocal_num_trees:  %i\n"
+                  "\tglobal_num_trees: %li\n",
+                  forest->first_local_tree,
+                  forest->last_local_tree,
+                  t8_forest_get_num_local_trees (forest),
+                  t8_forest_get_num_global_trees (forest));
+
+  t8_productionf ("The adapted forest with one empty tree:\n"
+                  "\tfirst_local_tree: %li\n"
+                  "\tlast_local_tree:  %li\n"
+                  "\tlocal_num_trees:  %i\n"
+                  "\tglobal_num_trees: %li\n",
+                  forest_adapt->first_local_tree,
+                  forest_adapt->last_local_tree,
+                  t8_forest_get_num_local_trees (forest_adapt),
+                  t8_forest_get_num_global_trees (forest_adapt));
+
+  t8_forest_unref (&forest_adapt);
+  t8_forest_unref (&forest);
+}
+
+int
+main (int argc, char **argv)
+{
+  char                usage[BUFSIZ];
+  /* brief help message */
+  int                 sreturnA = snprintf (usage, BUFSIZ,
+                                           "Usage:\t%s <OPTIONS>\n\t%s -h\t"
+                                           "for a brief overview of all options.",
+                                           basename (argv[0]),
+                                           basename (argv[0]));
+
+  char                help[BUFSIZ];
+  /* long help message */
+  int                 sreturnB = snprintf (help, BUFSIZ,
+                                           "We create a forest with a strip of quad trees.\n"
+                                           "One tree of this strip does not contain any elements.\n\n%s\n",
+                                           usage);
+
+  if (sreturnA > BUFSIZ || sreturnB > BUFSIZ) {
+    /* The usage string or help message was truncated */
+    /* Note: gcc >= 7.1 prints a warning if we 
+     * do not check the return value of snprintf. */
+    t8_debugf
+      ("Warning: Truncated usage string and help message to '%s' and '%s'\n",
+       usage, help);
+  }
+
+  int                 mpiret = sc_MPI_Init (&argc, &argv);
+  SC_CHECK_MPI (mpiret);
+
+  sc_init (sc_MPI_COMM_WORLD, 1, 1, NULL, SC_LP_ESSENTIAL);
+  t8_init (SC_LP_DEFAULT);
+
+  /* Parameter for t8_construct_fractal and command line */
+  t8_locidx_t         num_trees;
```

Comment:
```suggestion
  t8_gloidx_t         num_trees;
```

Reply:
It has to stay `t8_locidx_t` because of `sc_options_add_* (**)`. But I cast them to `t8_gloidx_t` when passing to `t8_strip_of_quads (*)`.